prime-router/docs/design/design/deduplication/deduplication.md

```diff
@@ -0,0 +1,248 @@
+    - Stamp Observations with condition codes
+  - During the Deduplication Workflow, the Convert Step will:
+    - Generate a hash for the item
+    - Compare this hash against those batched in the same submitted report
+    - Compare this hash with others stored in the database
+    - Create appropriate action logs and Azure Events if any duplicates were detected
+  - After the Deduplication Workflow occurs, the Convert Step will perform necessary pipeline functions:
+    - Apply sender schema transform
+    - Make a report object with item lineage
+      - **This will now include the newly generated (or null) hash**
+    - Create a route event
+    - Upload to blobstore
+    - Update the database
+    - Add the report to the route destination filter step queue
+
+### Other Updates
+- Sender Settings: The Universal Pipeline will use the existing `allowDuplicates` sender setting. 
+  - By default, the deduplication workflow is enabled for UP senders (`allowDuplicates` = false). Individual receivers will set `allowDuplicates` to `true` to opt out.
+  - The abstract Sender class [should not initialize](https://github.com/CDCgov/prime-reportstream/blob/f48d719b876859169deb0360487f63965d8be5a0/prime-router/src/main/kotlin/Sender.kt#L60) `allowDuplicates`.
+  - The UniversalPipelineSender class [should default to false](https://github.com/CDCgov/prime-reportstream/blob/f48d719b876859169deb0360487f63965d8be5a0/prime-router/src/main/kotlin/Sender.kt#L196).
+- Receiver Step: There is stubbed code to be removed here: The SumissionReceiver refers to the once theorized area where UP deduplication would be invoked. ([SubmissionReceiver.kt#L284-L292](https://github.com/CDCgov/prime-reportstream/blob/0c5e0b058e35e09786942f2c8b41c1d67a5b1d16/prime-router/src/main/kotlin/SubmissionReceiver.kt#L284-L292))
+- Database Changes: Several changes will need to take place to improve efficiency. **Over 90% of the values stored in item_lineage.item_hash are not relevant to either CP or UP deduplication.**
+  - Immediate change: Modify item_lineage.item_hash to accept NULL values.
+    - Reasoning: 90%+ of item_lineage table rows have no need to store the item_hash. Allowing null values will reduce updates to the index and indexing times for the table. 
+  - Long term change: Modify the index on item_lineage to a partial index, on only those values which are not null.
+    - Caveat: Barring a significant database wipe of out of date information, this would likely result in a significant _decrease_ in database efficiency.
+    - A hybrid plan would be to begin storing NULL values. After older values have been culled, and/or enough NULL values have accumulated in the table, then change the index to a partial index. 
+- Database: Investigation will need to happen on how to best purge out of date records on the item_lineage table. item_lineage.created_at will be used for this. This is outside the scope for this design/implementation.
```

Comment:
If there is an index on created_at performance should not be affected that much if we just limit our select query by created_at?

Reply:
Correct, and this was more aimed at the original notion of periodically purging the tables. I've removed this line as it isn't really relevant to the design here and added an "investigation results" bullet point under technical considerations with a brief summary of the db efficiency investigation and how created_at was relevant to that.